x-pack/plugins/observability/public/pages/alert_details/components/header_actions.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { noop } from 'lodash';
+import { CaseAttachmentsWithoutOwner } from '@kbn/cases-plugin/public/types';
+import { CommentType } from '@kbn/cases-plugin/common';
+import { EuiButton, EuiButtonEmpty, EuiFlexGroup, EuiPopover, EuiText } from '@elastic/eui';
+import { ALERT_RULE_UUID, ALERT_UUID } from '@kbn/rule-data-utils';
+
+import { useKibana } from '../../../utils/kibana_react';
+import { useFetchRule } from '../../../hooks/use_fetch_rule';
+import { ObservabilityAppServices } from '../../../application/types';
+import { TopAlert } from '../../alerts';
+
+export interface HeaderActionsProps {
+  alert: TopAlert | null;
+  externalConnector?: {
+    name: string;
+    onViewInExternalApp: (alertId: string) => void;
+  };
+}
+
+export function HeaderActions({ alert, externalConnector }: HeaderActionsProps) {
+  const {
+    http,
+    cases: {
+      hooks: { getUseCasesAddToExistingCaseModal },
+    },
+    triggersActionsUi: { getEditAlertFlyout, getRuleSnoozeModal },
+  } = useKibana<ObservabilityAppServices>().services;
+
+  const { rule, reloadRule } = useFetchRule({
+    http,
+    ruleId: alert?.fields[ALERT_RULE_UUID] || '',
+  });
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+  const [ruleConditionsFlyoutOpen, setRuleConditionsFlyoutOpen] = useState<boolean>(false);
+  const [snoozeModalOpen, setSnoozeModalOpen] = useState<boolean>(false);
+
+  const selectCaseModal = getUseCasesAddToExistingCaseModal();
+
+  const handleTogglePopover = () => setIsPopoverOpen(!isPopoverOpen);
+  const handleClosePopover = () => setIsPopoverOpen(false);
+
+  const attachments: CaseAttachmentsWithoutOwner =
+    alert && rule
+      ? [
+          {
+            alertId: alert?.fields[ALERT_UUID] || '',
+            index: '.internal.alerts-observability.metrics.alerts-*',
```

Comment:
The API that we are using to fetch alert data can be possibly modified to return index as well. Currently it returns `_source`. The `alert.hits.hits[0]` also has `_index` property. 
@simianhacker @fkanout what do you think?

https://github.com/elastic/kibana/blob/2d8eb0eed6a8325d7f630cd8f539055e306d371a/x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts#L498

Reply:
@simianhacker Not related to the scope of PR, but there is a comment above this line of code "move away from pulling data from _source in the future". What is the other way if we don't want to use "_source"? Is this implemented somewhere else?

Reply:
> The API that we are using to fetch alert data can be possibly modified to return index as well. Currently it returns _source. The alert.hits.hits[0] also has _index property. @simianhacker @fkanout what do you think?

What would we use the `_index` field for?

> What is the other way if we don't want to use "_source"? Is this implemented somewhere else to have a look?

The alternative is to use the `fields` API (which is a setting on the request). I think the `fields` API becomes more advantageous when the index is not setup to store the `_source` and only has `fieldData`. I believe the fields API also resolves runtime fields as well (which we are not using).

Reply:
@simianhacker We need the index name here - `index: '.internal.alerts-observability.metrics.alerts-*',`
Currently it is hardcoded.

Reply:
Then yes, we need to get the `_index` field as well.